File: Day 2 - 11072026/studentManagementSystem/src/main/java/com/example/studentManagementSystem/service/StudentService.java
package com.example.studentManagementSystem.service;

import com.example.studentManagementSystem.model.Student;
import com.example.studentManagementSystem.repository.StudentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class StudentService {

    @Autowired
    private StudentRepository studentRepository;

    public List<Student> getAllStudents(){
        return studentRepository.findAll();
    }

    public Student getStudentById(Long id) {
        return studentRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Student Not Found"));
    }
    public Student addStudent(Student student) {
        return studentRepository.save(student);
    }

    public Student updateStudent(Student student, Long id){
        Student old = getStudentById(id);

        old.setFirstName(student.getFirstName());
        old.setLastName(student.getLastName());
        old.setEmail(student.getEmail());
        old.setCourse(student.getCourse());
        old.setPhone(student.getPhone());
        old.setDepartment(student.getDepartment());
        old.setYear(student.getYear());

        return studentRepository.save(old);
    }

    public void deleteStudent(Long id) {
        studentRepository.delete(getStudentById(id));
    }

    public List<Student> searchStudents(String firstName) {
        return studentRepository.findByFirstNameContainingIgnoreCase(firstName);
    }

}

File: Day 2 - 11072026/Frontend/frontend/src/services/studentService.js
import axios from "axios";

const API = "http://localhost:8080/students";

export const getStudents = () => axios.get(API);

export const getStudent = (id) => axios.get(`${API}/${id}`);

export const addStudent = (student) => axios.post(API, student);

export const updateStudent = (id, student) =>
  axios.put(`${API}/${id}`, student);

export const deleteStudent = (id) =>
  axios.delete(`${API}/${id}`);

export const searchStudent = (firstName) =>
    axios.get(`${API}/search?firstName=${firstName}`);